### PR TITLE
Adopt `0.18.1` SDK

### LINKED ISSUE #2070: Integrate `0.18.x` SDK
### Problem

We can reduce FCM memory usage by migrating to the new `MerkleMap` in the `0.18.x` SDK.

### Solution

Migrate to the `0.18.x` SDK.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Note to reviewers**:
- All nightly regression tests look comparable to `master` except the testnet migrations, which fail because the `FCMap -> MerkleMap` migration takes longer than we allow; if these cannot be done in parallel, we may need to extend the time limit in those tests.
- Leaving _.circleci/config.yml_ changes in for now to keep running nightly regression tests.

**Description**:
- Migrates from `FCMap` to `MerkleMap`.
- Uses non-Merkle key types for all `MerkleMap`s---either `EntityNum`, `EntityNumPair`, or `String`.
    * The new [`StaticPropertiesHolder` singleton](https://github.com/hashgraph/hedera-services/blob/38d0bc85a84464d4eb8837021efe7b4a6a9ba772/hedera-node/src/main/java/com/hedera/services/context/properties/StaticPropertiesHolder.java#L35) supports easy translation from an entity number to a full `<shard>.<realm>.<number>` entity id.
- Previous Merkle types used for FCM values---e.g. `MerkleEntityId`---are retained **for migration only**; if any other use of these types was retained, this was unintentional and likely a bug.
- The new `SwirldState.migrate()` method can only run _after_ `SwirldState.init()` due to the `BinaryObjectStore` initialization lifecycle; hence the full `SwirldState.init()` process must be deferred until `SwirldState.migrate()`.

**Related issue(s)**:
- Closes #2070

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/sigs/metadata/lookups/DefaultTopicLookup.java [hunk 1]
```diff
@@ -21,31 +21,32 @@
...
-				new TopicSigningMetadata(
-						topic.hasAdminKey() ? topic.getAdminKey() : null,
-						topic.hasSubmitKey() ? topic.getSubmitKey() : null));
+		if (topic == null || topic.isDeleted()) {
+			return SafeLookupResult.failure(INVALID_TOPIC);
+		} else {
+			final var effAdminKey = topic.hasAdminKey() ? topic.getAdminKey() : null;
+			final var effSubmitKey = topic.hasSubmitKey() ? topic.getSubmitKey() : null;
+			return new SafeLookupResult<>(new TopicSigningMetadata(effAdminKey, effSubmitKey));
+		}
```
Comment (ljianghedera) [inline hedera-node/src/main/java/com/hedera/services/sigs/metadata/lookups/DefaultTopicLookup.java:50]: This is much clearer!

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleOptionalBlob.java [hunk 2]
```diff
@@ -28,14 +28,19 @@
...
 import java.util.Objects;
 import java.util.function.Supplier;
 
-public class MerkleOptionalBlob extends AbstractMerkleLeaf implements MerkleExternalLeaf {
-	static final int MERKLE_VERSION = BinaryObject.ClassVersion.ORIGINAL;
+public class MerkleOptionalBlob extends AbstractMerkleLeaf implements MerkleExternalLeaf, Keyed<String> {
+	static final int PRE_RELEASE_0180_VERSION = 1;
+	static final int RELEASE_0180_VERSION = 2;
+
+	static final int CURRENT_VERSION = RELEASE_0180_VERSION;
```
Comment (ljianghedera) [inline hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleOptionalBlob.java:41]: nit: Is `MERKEL_VERSION` semantically clearer than `CURRENT_VERSION`?

File: hedera-node/src/main/java/com/hedera/services/state/merkle/internals/BitPackUtils.java [hunk 3]
```diff
@@ -90,6 +119,18 @@ public static int codeFromNum(long num) {
 		return (int) num;
```
Comment (ljianghedera) [inline hedera-node/src/main/java/com/hedera/services/state/merkle/internals/BitPackUtils.java:119]: If the `num` value IS `0xFFFFFFFF` here, will the return value be overflowed?

File: hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java [hunk 4]
```diff
@@ -160,9 +159,10 @@ public long currentMintedNfts() {
 	public TokenRelationship loadTokenRelationship(Token token, Account account) {
 		final var tokenId = token.getId();
 		final var accountId = account.getId();
-		final var key = new MerkleEntityAssociation(
-				accountId.getShard(), accountId.getRealm(), accountId.getNum(),
-				tokenId.getShard(), tokenId.getRealm(), tokenId.getNum());
+//		final var key = new MerkleEntityAssociation(
+//				accountId.getShard(), accountId.getRealm(), accountId.getNum(),
+//				tokenId.getShard(), tokenId.getRealm(), tokenId.getNum());
```
Comment (ljianghedera) [inline hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java:164]: Can these 3 lines be removed?

**Review by ljianghedera — DISMISSED** [review]
Looks great!

File: hedera-node/src/main/java/com/hedera/services/sigs/metadata/lookups/DefaultTopicLookup.java (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/sigs/metadata/lookups/DefaultTopicLookup.java:50]: Thanks Sonar :D

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleOptionalBlob.java (hunk 2 above)
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleOptionalBlob.java:41]: I'm liking `CURRENT_VERSION` more these days...what do you think?

File: hedera-node/src/main/java/com/hedera/services/state/merkle/internals/BitPackUtils.java (hunk 3 above)
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/state/merkle/internals/BitPackUtils.java:119]: The "codes" are (signed) `int`s in the range `[Integer.MIN_VALUE, Integer.MAX_VALUE]`

File: hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java (hunk 4 above)
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java]: Removed

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleOptionalBlob.java (hunk 2 above)
Comment (ljianghedera) [reply hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleOptionalBlob.java:41]: Yeah, it makes sense after going through the usages.

File: hedera-node/src/main/java/com/hedera/services/state/merkle/internals/BitPackUtils.java (hunk 3 above)
Comment (ljianghedera) [reply hedera-node/src/main/java/com/hedera/services/state/merkle/internals/BitPackUtils.java:119]: I see. In reality for this case, there will never be a chance for the overflow to happen.

**Review by ljianghedera — APPROVED** [review]

**Pull request merged**